### PR TITLE
(0012894) Change controller_init_setup lambda to VPC

### PR DESCRIPTION
**Pull request by zhiqiangjin**

### CODE REVIEW

File: aviatrix-controller-build/outputs.tf [hunk 1]
```diff
@@ -5,3 +5,15 @@ output "private_ip" {
 output "public_ip" {
   value = "${aws_eip.controller_eip.*.public_ip[0]}"
 }
+
+output "vpc" {
```
Comment (arvindsree) [inline aviatrix-controller-build/outputs.tf]: change to vpc_id

File: aviatrix-controller-build/outputs.tf [hunk 2]
```diff
@@ -5,3 +5,15 @@ output "private_ip" {
...
+  value = var.vpc
+}
+
+output "subnet" {
```
Comment (arvindsree) [inline aviatrix-controller-build/outputs.tf]: subnet_id

File: aviatrix-controller-initialize/main.tf [hunk 3]
```diff
@@ -18,9 +22,37 @@ resource "aws_iam_role" "iam_for_lambda" {
...
+    }
+  ]
+}
+EOF
+}
+
 resource "aws_iam_role_policy_attachment" "attach-policy" {
   role       = aws_iam_role.iam_for_lambda.name
-  policy_arn = "arn:aws:iam::aws:policy/CloudWatchAgentServerPolicy"
+  policy_arn = aws_iam_policy.lambda-policy.arn
```
Comment (arvindsree) [inline aviatrix-controller-initialize/main.tf]: Does the destroy work with the regular CloudWatchAgentServerPolicy policy?

File: aviatrix-controller-initialize/security_group.tf [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  description = "Aviatrix - Controller Security Group 2"
+  vpc_id      = var.vpc
+
+  tags = {
+    Name      = "${local.name_prefix}AviatrixSecurityGroup2"
+    Createdby = "Terraform+Aviatrix2"
+  }
+}
+
+resource "aws_security_group_rule" "ingress_rule" {
```
Comment (arvindsree) [inline aviatrix-controller-initialize/security_group.tf]: Is this ingress rule required?

File: aviatrix-controller-initialize/security_group.tf [hunk 5]
```diff
@@ -0,0 +1,28 @@
+resource "aws_security_group" "AviatrixSecurityGroup2" {
+  name        = "${local.name_prefix}AviatrixSecurityGroup2"
+  description = "Aviatrix - Controller Security Group 2"
```
Comment (arvindsree) [inline aviatrix-controller-initialize/security_group.tf]: Aviatrix Lambda secuyrity grroup

File: aviatrix-controller-initialize/main.tf (hunk 3 above)
Comment (zhiqiangjin) [reply aviatrix-controller-initialize/main.tf]: Regular CloudWatchAgentServerPolicy policy does not have permissions to call CreateNetworkInterface on EC2, so changes are needed.

Comment (arvindsree) [reply aviatrix-controller-initialize/main.tf]: ok

**Comment by arvindsree** [comment]
change PR title

File: aviatrix-controller-initialize/security_group.tf (hunk 4 above)
Comment (zhiqiangjin) [reply aviatrix-controller-initialize/security_group.tf]: This ingress rule is not required. Will remove it.

File: aviatrix-controller-initialize/variables.tf [hunk 6]
```diff
@@ -24,5 +24,16 @@ variable "customer_license_id" {
 }
 
 variable "controller_launch_wait_time" {
-  default = 180
+  default = 480
```
Comment (arvindsree) [inline aviatrix-controller-initialize/variables.tf]: this is too high. Revert it back

File: aviatrix-controller-initialize/variables.tf [hunk 7]
```diff
@@ -24,5 +24,16 @@ variable "customer_license_id" {
...
 }
+
+variable "vpc" {}
+
+variable "subnet" {}
+
+variable "incoming_ssl_cidr" {
```
Comment (arvindsree) [inline aviatrix-controller-initialize/variables.tf]: remove this as well

File: aviatrix-controller-initialize/variables.tf [hunk 8]
```diff
@@ -26,3 +26,7 @@ variable "customer_license_id" {
 variable "controller_launch_wait_time" {
   default = 180
 }
+
+variable "vpc" {}
+
+variable "subnet" {}
```
Comment (arvindsree) [inline aviatrix-controller-initialize/variables.tf]: vpc_id and subnet_id

**Pull request merged**